src/Widgets/SharePopover.vala

```diff
@@ -107,59 +82,24 @@ public class SharePopover : Gtk.Popover {
             }
             hide ();
         });
+    }
 
-        facebook_button.clicked.connect (() => {
-            try {
-                AppInfo.launch_default_for_uri ("https://www.facebook.com/sharer/sharer.php?u=%s".printf (uri), null);
-            } catch (Error e) {
-                warning ("%s", e.message);
-            }
-            hide ();
-        });
-
-        google_button.clicked.connect (() => {
-            try {
-                AppInfo.launch_default_for_uri ("https://plus.google.com/share?url=%s".printf (uri), null);
-            } catch (Error e) {
-                warning ("%s", e.message);
-            }
-            hide ();
-        });
-
-        twitter_button.clicked.connect (() => {
-            try {
-                AppInfo.launch_default_for_uri ("http://twitter.com/home/?status=%s %s".printf (body, uri), null);
-            } catch (Error e) {
-                warning ("%s", e.message);
-            }
-            hide ();
-        });
-
-        reddit_button.clicked.connect (() => {
-            try {
-                AppInfo.launch_default_for_uri ("http://www.reddit.com/submit?title=%s&url=%s".printf (body, uri), null);
-            } catch (Error e) {
-                warning ("%s", e.message);
-            }
-            hide ();
-        });
-
-        tumblr_button.clicked.connect (() => {
-            try {
-                AppInfo.launch_default_for_uri ("https://www.tumblr.com/share/link?url=%s".printf (uri), null);
-            } catch (Error e) {
-                warning ("%s", e.message);
-            }
-            hide ();
-        });
-
-        telegram_button.clicked.connect (() => {
+    Gtk.Widget make_share_button(string name, string uri_template, string icon_name) {
```

Comment:
Instead of a method to return a widget this should be a `Gtk.Button` subclass

Reply:
Thank you for your code review. I apologize for being a bit too sloppy with the code. I'll see that it doesn't happen again.

Subclassing widgets, however, seems to be overkill unless you intend use them elsewhere. I'll post a PR with a polished version of the code done with helper methods, then if you'll still want widgets done I'll comply.

As for the main part of code, I'll put it on hold for a couple of days until it becomes evidently useful.

Reply:
Hey no worries! Can't expect you to know the code style until you've seen the guide :) I've left a comment on the other PR, but I'd like to ask some other folks to weigh in and see what the consensus is